Lib/test/test_asyncio/test_streams.py

```diff
@@ -801,6 +801,25 @@ def test_streamreaderprotocol_constructor_use_global_loop(self):
         self.assertEqual(cm.warnings[0].filename, __file__)
         self.assertIs(protocol._loop, self.loop)
 
+    def test_streamreaderprotocol_task_reference(self):
+        # See http://bugs.python.org/issue46309
+        messages = []
+        self.loop.set_exception_handler(lambda loop, ctx: messages.append(ctx))
+
+        async def callback(*args):
+            await self.loop.create_future()
+
+        async def test():
+            reader = asyncio.StreamReader()
+            protocol = asyncio.StreamReaderProtocol(reader, callback)
+            transport = mock.Mock()
+            protocol.connection_made(transport)
+            await asyncio.sleep(0)
+            gc.collect()
```

Comment:
```suggestion
            gc.collect(1)
```

Full collection will cause all test items to go into gen2 which is both slow and not required.